src/pages/workspace/WorkspaceSettingsPage.js

```diff
@@ -59,20 +63,22 @@ const defaultProps = {
 function WorkspaceSettingsPage({policy, currencyList, windowWidth, route}) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
+    const currencyParam = lodashGet(route, 'params.currency', '').toUpperCase();
+    const currencyCode = CurrencyUtils.isValidCurrencyCode(currencyParam) ? currencyParam : policy.outputCurrency;
 
-    const formattedCurrency = !_.isEmpty(policy) && !_.isEmpty(currencyList) ? `${policy.outputCurrency} - ${currencyList[policy.outputCurrency].symbol}` : '';
+    const formattedCurrency = !_.isEmpty(policy) && !_.isEmpty(currencyList) ? `${currencyCode} - ${currencyList[currencyCode].symbol}` : '';
 
     const submit = useCallback(
         (values) => {
             if (policy.isPolicyUpdating) {
                 return;
             }
 
-            Policy.updateGeneralSettings(policy.id, values.name.trim(), policy.outputCurrency);
+            Policy.updateGeneralSettings(policy.id, values.name.trim(), currencyCode);
             Keyboard.dismiss();
-            Navigation.goBack(ROUTES.WORKSPACE_INITIAL.getRoute(policy.id));
+            Navigation.navigate(ROUTES.WORKSPACE_INITIAL.getRoute(policy.id));
```

Comment:
I change the goBack with navigate because if we open currency selection page by url for example: `https://dev.new.expensify.com:8082/workspace/WORKSPACEID/settings/currency?currency=NZD` then click a currency then click save it will go back to currency selection list page and not to workspace setting page.